backend/src/main/java/com/elshop/backend/common/S3Service.java

```diff
@@ -0,0 +1,82 @@
+package com.elshop.backend.common;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+@Component
+public class S3Service {
+    private final AmazonS3 s3;
+
+    private final String bucketName;
+
+    public S3Service(@Value("${cloud.aws.bucket}") String bucketName, AmazonS3 s3) {
+        this.s3 = s3;
+        this.bucketName = bucketName;
+    }
+
+    public List<String> uploadMultipleFiles(MultipartFile[] images, String type, String entityId) {
+        return Arrays.stream(images)
+                .map(image -> uploadFile(image, type, entityId))
+                .toList();
+    }
+
+    public String uploadFile(MultipartFile file, String type, String entityId) {
+        String originalFilename = file.getOriginalFilename();
+        String bucketKey;
+        try {
+            File imageFile = convertMultiPartToFile(file);
+            bucketKey = String.format("%s/%s/%s", type, entityId, originalFilename);
+            s3.putObject(bucketName, bucketKey, imageFile);
+            Files.delete(imageFile.toPath());
+            return s3.getUrl(bucketName, bucketKey).toString();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
Wenn möglich vermeide, eine so allgemeine Exception wie eine `RuntimeException` zu werfen. Darüber hinaus wäre es schön, den anderen Entwickelnden mehr Hinweise zu geben, was hier schief läuft.
```suggestion
            throw new IllegalStateException("Error reading the multipartfile of the uploaded image", e);
```